### PR TITLE
[Bugfix] history support middleware allowed to be disabled

### LINKED ISSUE #4557: History support middleware cannot be disabled
It's nice that history support is middleware, however in order to support my use case, I need to write my own version of this plugin.

In my config I set historySupportMiddlewareEnabled to false

Unforutnately, line 18 of history-support's index.js reads:

    return ['auto', 'history'].indexOf(locationType) !== -1 || historySupportMiddlewareEnabled;

The middleware loads anyway.

### PR DESCRIPTION
**Pull request by jasonmit**
Fixes  #4557

### CODE REVIEW

File: lib/tasks/server/middleware/history-support/index.js [hunk 1]
```diff
@@ -13,9 +13,13 @@ function HistorySupportAddon(project) {
 HistorySupportAddon.prototype.shouldAddMiddleware = function(environment) {
   var config = this.project.config(environment);
   var locationType = config.locationType;
-  var historySupportMiddlewareEnabled = config.historySupportMiddleware;
+  var historySupportMiddleware = config.historySupportMiddleware;
```
Comment (jasonmit) [inline lib/tasks/server/middleware/history-support/index.js]: If passed, it takes priority over locationType

**Comment by rwjblue** [comment]
Looks good to me once the failing tests are fixed.

**Comment by stefanpenner** [comment]
Lgtm

**Comment by jasonmit** [comment]
@rwjblue @stefanpenner all fixed - thanks

**Comment by stefanpenner** [comment]
sg

**Comment by jasonmit** [comment]
The other failures are likely not related to my changes, at least not that I can tell by.

**Comment by stefanpenner** [comment]
ya looks like a low timeout for one of the acceptance tests. Gave it a kick

**Comment by stefanpenner** [comment]
@homu r+

**Comment by rwjblue** [comment]
@homu retry

**Comment by rwjblue** [comment]
Failure in Travis is the issue from http-parser that was recently fixed on master.  Merging...

**Pull request merged**